tests/functional/macros/test_macros.py

```diff
@@ -0,0 +1,337 @@
+        run_dbt(["deps"])
+        results = run_dbt()
+        assert len(results) == 6
+
+        check_relations_equal(project.adapter, ["expected_dep_macro", "dep_macro"])
+        check_relations_equal(project.adapter, ["expected_local_macro", "local_macro"])
+
+
+class TestInvalidMacros(object):
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "dep_macro.sql": models__dep_macro,
+            "local_macro.sql": models__local_macro,
+            "ref_macro.sql": models__ref_macro,
+        }
+
+    def test_invalid_macro(self, project):
+        run_dbt(expect_pass=False)
+
+
+class TestAdapterMacroNoDestination(object):
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {"model.sql": models__with_undefined_macro}
+
+    @pytest.fixture(scope="class")
+    def macros(self):
+        return {"my_macros.sql": macros__no_default_macros}
+
+    def test_invalid_macro(self, project):
+        with pytest.raises(dbt.exceptions.CompilationException) as exc:
+            run_dbt()
+
+        assert "In dispatch: No macro named 'dispatch_to_nowhere' found" in str(exc.value)
+
+
+class TestMacroOverrideBuiltin(object):
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {"model.sql": models__override_get_columns_macros}
+
+    @pytest.fixture(scope="class")
+    def macros(self):
+        return {"macros.sql": macros__override_get_columns_macros}
+
+    def test_overrides(self, project):
+        # the first time, the model doesn't exist
+        run_dbt()
+        run_dbt()
+
+
+class TestMacroOverridePackage(object):
+    """
+    The macro in `override-postgres-get-columns-macros` should override the
+    `get_columns_in_relation` macro by default.
+    """
+
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {"model.sql": models__override_get_columns_macros}
+
+    @pytest.fixture(scope="class")
+    def macros(self):
+        return {"macros.sql": macros__package_override_get_columns_macros}
+
+    def test_overrides(self, project):
+        # the first time, the model doesn't exist
+        run_dbt()
+        run_dbt()
+
+
+class TestMacroNotOverridePackage(object):
+    """
+    The macro in `override-postgres-get-columns-macros` does NOT override the
+    `get_columns_in_relation` macro because we tell dispatch to not look at the
+    postgres macros.
+    """
+
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {"model.sql": models__override_get_columns_macros}
+
+    @pytest.fixture(scope="class")
+    def macros(self):
+        return {"macros.sql": macros__package_override_get_columns_macros}
+
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "dispatch": [{"macro_namespace": "dbt", "search_order": ["dbt"]}],
+        }
+
+    def test_overrides(self, project):
+        # the first time, the model doesn't exist
```

Comment:
Pretty sure this is due to:

> {% if execute and result != 'a string' %}
>   {% do exceptions.raise_compiler_error('overriding get_columns_in_relation failed') %}
> {% endif %}

The model this test calls will build and then because it now exists, this exception will throw. My understanding is we want _that_ error to be caught. `execute` makes this sketchy since that's a complicated flag.